### PR TITLE
DEPR: row-based indexing in DataFrame.__getitem__

### PR DESCRIPTION
**Pull request by jbrockmendel**
Discussed recently in #9595.  The two tests this edits are the only two that go down this path AFAICT.

### CODE REVIEW

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
Let's first discuss this a bit more, as this is a huge change. (I am personally -1 on changing this I think)

**Comment by jbrockmendel** [comment]
@jorisvandenbossche discussion notwithstanding, can i get your help in troubleshooting the doc build?  i put in "okwarnings" to fix the previous set of failures, not clear on why dft_minute is now undefined

File: doc/source/user_guide/timeseries.rst [hunk 1]
```diff
@@ -677,6 +677,7 @@ If index resolution is second, then the minute-accurate timestamp gives a
 If the timestamp string is treated as a slice, it can be used to index ``DataFrame`` with ``[]`` as well.
 
 .. ipython:: python
+   :okwarning:
 
     dft_minute = pd.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]},
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/timeseries.rst]: I think the whitespace alignment is off here (`:okwarning:` and `dft_minute...` should be vertically aligned)

File: doc/source/user_guide/timeseries.rst [hunk 2]
```diff
@@ -689,7 +691,11 @@ If the timestamp string is treated as a slice, it can be used to index ``DataFra
 
    To *always* have unambiguous selection, whether the row is treated as a slice or a single selection, use ``.loc``.
 
+   As of version 1.1.0, row-based indexing is deprecated.  Users should always use ``.loc``
+   for label-based row indexing.
+
    .. ipython:: python
+      :okwarning:
```
Comment (TomAugspurger) [inline doc/source/user_guide/timeseries.rst]: Does this cause a warning?

**Review by TomAugspurger — COMMENTED** [review]
@jbrockmendel can you give examples of what user-facing behavior is being deprecated? I'd be in favor of deprecating fallback indexing for partial string slicing dataframes with datetime indexes

```python
In [18]: df = pd.DataFrame({"A": [1, 2]}, index=pd.date_range('2000', periods=2))

In [19]: df['2000']
Out[19]:
            A
2000-01-01  1
2000-01-02  2
```

**Comment by jorisvandenbossche** [comment]
Repeating what I said above again: please open an issue first to discuss this.

**Comment by jreback** [comment]
iirc there is an issue about this (though woudl have to search)

**Comment by jorisvandenbossche** [comment]
The issue Brock mentioned is #9595, but that is much broader (about ``__getitem`` in general), so if that is the one you are thinking about, I would propose to create a new, specific issue for this topic

**Comment by jreback** [comment]
> The issue Brock mentioned is #9595, but that is much broader (about `__getitem` in general), so if that is the one you are thinking about, I would propose to create a new, specific issue for this topic

i believe there is a specific issue this was discussed long ago

File: doc/source/user_guide/timeseries.rst (hunk 2 above)
Comment (jbrockmendel) [reply doc/source/user_guide/timeseries.rst]: i took the kitchen-sink approach to getting the docbuild passing, will figure out which of these can be trimmed away

**Comment by jbrockmendel** [comment]
> can you give examples of what user-facing behavior is being deprecated? I'd be in favor of deprecating fallback indexing for partial string slicing dataframes with datetime indexes

Exactly the example you gave.  The relevant cases are where we have all of the following:
- a DataFrame `df`
- `df.index` is either `DatetimeIndex` or `PeriodIndex`
- `key` is a date-like string
- `key not in df.columns`

(these conditions are basically psuedo-code for `convert_to_index_sliceable`)

**Comment by jbrockmendel** [comment]
closing to clear the queue, will reivisit

**Pull request closed without merging**